### PR TITLE
refactor: make api connection a class function

### PR DESCRIPTION
**Pull request by CFenner**

### CODE REVIEW

File: custom_components/casambi_bt/__init__.py [hunk 1]
```diff
@@ -27,28 +27,22 @@
...
-        hass, conf[CONF_ADDRESS], conf[CONF_PASSWORD]
-    )
-
-    if not casa_api:
-        return False
-
-    casa_api.casa.registerUnitChangedHandler(casa_api._unit_changed_handler)
+    api = CasambiApi(hass, entry.data[CONF_ADDRESS], entry.data[CONF_PASSWORD])
+    await api.connect()
+    api.register_unit_changed_handler()
```
Comment (CFenner) [inline custom_components/casambi_bt/__init__.py:32]: This was/is only applied on setup, does it need to be called again on reconnect?

Comment (lkempf) [reply custom_components/casambi_bt/__init__.py:32]: No as long as the Casmabi object doesn't change this is fine. The internal callbacks that change with a new connection get abstracted away in the library.

File: custom_components/casambi_bt/__init__.py [hunk 2]
```diff
@@ -69,58 +63,50 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
...
-        self.casa = casa
         self.hass = hass
         self.address = address
         self.password = password
+        self.casa: Casambi = Casambi(hass.helpers.httpx_client.get_async_client())
 
-        self._register_bluetooth_callback()
-
-    def _register_bluetooth_callback(self):
+    def register_bluetooth_callback(self):
```
Comment (lkempf) [inline custom_components/casambi_bt/__init__.py:79]: Is there a reason why the _ was removed? As far as I can see this method is still only used internally.

Comment (CFenner) [reply custom_components/casambi_bt/__init__.py:79]: I thought this is the convention for internal methonds

Comment (CFenner) [reply custom_components/casambi_bt/__init__.py:79]: Ah, right, it is used only internal now again. So will change it back.

File: custom_components/casambi_bt/__init__.py (hunk 1 above)
Comment (CFenner) [reply custom_components/casambi_bt/__init__.py:32]: Does it require a connection? Otherwise this could go to the init function.

Comment (lkempf) [reply custom_components/casambi_bt/__init__.py:32]: No the internal callback required for this callback to fire only gets subscribed to in connect. So register_unit_changed_handler needs to be called before the call to connect (not the order it currently is in)

Comment (CFenner) [reply custom_components/casambi_bt/__init__.py:32]: So, should I change it or leave it?

**Comment by lkempf** [comment]
Ok that's not quite what I wanted :smile:

**Comment by lkempf** [comment]
I think the easiest way forward is to abandon this PR and simply commit the final changed file

**Pull request closed without merging**